domene/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/domene/behandling/Førstegangsbehandling.kt

```diff
@@ -190,8 +191,25 @@ data class Førstegangsbehandling(
         if (tilstand == BehandlingTilstand.TIL_BESLUTTER) {
             // TODO Gjør noe ekstra
         }
-        return this.copy(tiltak = tiltak.oppdaterTiltak(nyeTiltak), tilstand = BehandlingTilstand.OPPRETTET)
-            .vilkårsvurder()
+
+        val vurderingsperiodeFraTiltak = nyeTiltak.vurderingsperiodeFraTiltak()
+        if (vurderingsperiodeFraTiltak != null && vurderingsperiodeFraTiltak != this.vurderingsperiode) {
+            // Vurderingsperioden må endres!
```

Comment:
Burde vi sjekke om vurderingsperioden tilstøter eller overlapper? Bør vi hente alle tiltakene etter søknadsfristen på nytt og bestemme den nye vurderingsperioden utifra dette? Hva hvis det er en mismatch mellom disse og det som er lagret på behandlingen?

Reply:
Vi burde (kanskje?) hente inn tiltak som både overlapper og tilstøter, men filtreringen av tiltak basert på vurderingsperiode gjøres ikke her, men i `BehandlingServiceImpl::oppdaterTiltak(behandlingId: BehandlingId, tiltak: List<Tiltak>)` når vi mottar "løsningen" på "hentTiltak-behovet". Så den funksjonen kan endres til å ta med tilstøtende tiltak også.

Men det andre du skriver er korrekt - når vurderingsperioden utvides, så må vi egentlig hente inn tiltak på nytt, for det kan være at tiltak som har blitt filtrert bort nå vil bli tatt med. (Dette kan jo i teorien fortsette en stund..) Dette ville jeg helst ha løst med en form for event-mekanisme, men det mååå vel ikke gjøres på den måten.. :)